server_test.go

```diff
@@ -595,10 +602,7 @@ func checkInProgressQueriesAtShutdownServer(t *testing.T, srv *Server, addr stri
 			t.Errorf("ResponseWriter.WriteMsg error: %s", err)
 		}
 	})
-	defer func() {
-		testShutdownNotify = nil
-		HandleRemove("example.com.")
-	}()
+	defer HandleRemove("example.com.")
```

Comment:
why remove the "nil-ing" of testShutdownNotify ? 
this nil status is still the one that is tested in Server.shutdownContext ...

Reply:
@fturib The nil-test in `ShutdownContext` is there because `testShutdownNotify` isn't populated except when testing, it's not relevant to the actual test. When I used a per-test `chan` before, I was tripped the race detector. Now I'm using a `sync.Cond` (set only in `init`) which doesn't need to be replaced, and can't be without tripped the race detector. It's also side-effect free, so there's no harm with `Broadcast` being called in `ShutdownContext`.

Reply:
ok.